tests/test_tokenizer.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+from transformers import GPT2Config, GPT2Tokenizer, GPT2TokenizerFast
+
+from generative_data_prep.__main__ import get_tokenizer
+
+
+def test_pretrained_tokenizer():
+    """Test the tokenizer function using a pretrained_tokenizer."""
+    tokenizer, model_config = get_tokenizer(
+        pretrained_tokenizer="gpt2", tokenizer_class=None, vocab_file=None, merges_file=None, special_tokens_dict=None
+    )
+    assert type(tokenizer) == GPT2TokenizerFast
+    assert type(model_config) == GPT2Config
+
+
+def test_tokenizer_class():
+    """Test the tokenizer function using a tokenizer_class with vocab and merge files."""
+    current_dir = os.getcwd()
+    vocab_file = os.path.join(current_dir, "tests", "gpt2_vocab_and_merge_files", "vocab.json")
+    merge_file = os.path.join(current_dir, "tests", "gpt2_vocab_and_merge_files", "merges.txt")
+    tokenizer, model_config = get_tokenizer(
+        pretrained_tokenizer=None,
+        tokenizer_class="gpt2",
+        vocab_file=vocab_file,
+        merges_file=merge_file,
+        special_tokens_dict=None,
+    )
+    assert type(tokenizer) == GPT2Tokenizer
+    assert type(model_config) == GPT2Config
+
+
+def test_tokenizer_class_failure():
+    """Test the tokenizer function using a tokenizer_class but with the
+    intention of failing due to missing vocab and merge files.
+    """
+    try:
+        _, _ = get_tokenizer(
+            pretrained_tokenizer=None,
+            tokenizer_class="gpt2",
+            vocab_file=None,
+            merges_file=None,
+            special_tokens_dict=None,
+        )
+    except ValueError:
+        assert True
+        return
+    assert False
+
+
+def test_tokenizer_class_not_implemented():
+    """Test the tokenizer function using a tokenizer_class but with the intention of
+    failing due to using a model type that isn't added to the TOKENIZER_CLASS.
+    """
+    current_dir = os.getcwd()
```

Comment:
Does this always work? Maybe you should get the directory using the __file__, in case the pytest is executed from a different directory? [link](https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory)